recipes/libbasisu/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+    description = "Basis Universal Supercompressed GPU Texture Codec"
+    homepage = "https://github.com/BinomialLLC/basis_universal"
+    topics = ("conan", "basis", "textures", "compression")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False],
+        "sse": [True, False],
+        "x64": [True, False],
+        "zstd": [True, False],
+        "no_iterator_debug_level": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+        "sse": True,
+        "x64": True,
+        "zstd": True,
+        "no_iterator_debug_level": True
+    }
+    short_paths = True
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _minimal_compiler_version(self) -> bool:
+        version = tools.Version(self.settings.compiler.version.value)
+        return (self.settings.compiler == "Visual Studio" and version >= "15") \
+            or (self.settings.compiler == "gcc" and version >= "5.4") \
+            or (self.settings.compiler == "clang" and version >= "3.9") \
+            or (self.settings.compiler == "apple-clang" and version >= "10")
+ 
+    def configure(self):
+        if not self._minimal_compiler_version():
+            raise ConanInvalidConfiguration("{} {} does not support Visual Studio <= 14 with shared:True".format(self.name, self.version))
+
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_X64"] = self.options.x64
+        self._cmake.definitions["SSE"] = self.options.sse
+        self._cmake.definitions["ZSTD"] = self.options.zstd
+        self._cmake.definitions["BASISU_NO_ITERATOR_DEBUG_LEVEL"] = self.options.no_iterator_debug_level
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _clear_vs_project_files(self):
+        tools.remove_files_by_mask(self._source_subfolder, "*.sln")
+        tools.remove_files_by_mask(self._source_subfolder, "*.vcxproj")
+        tools.remove_files_by_mask(self._source_subfolder, "*.vcxproj.filters")
+        
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        self._clear_vs_project_files()
```

Comment:
Is this clearing necessary? Does it bother the cmake build system?
If not, why remove it? I'ld just keep it.